packages/legend-studio/src/components/editor/edit-panel/uml-editor/AssociationEditor.tsx

```diff
@@ -76,13 +76,22 @@ const AssociationPropertyBasicEditor = observer(
     association: Association;
     property: Property;
     selectProperty: () => void;
+    checkDuplicateValue: (val: string) => void;
```

Comment:
I'm not so convinced that we need to pass this in as a prop like this. We can construct this function right in the body of `AssociationPropertyBasicEditor`. Also, I am thinking about the general strategy that you used, it's not bad to do a warning, but kinda disruptive in terms of UX, imagine, we already have a property called `prop1`, if user is creating `prop12`, while typing, they would get this notification, that's not great. So instead, let's make this input box red, we can discuss the styling later, but you can follow some example we have (see `ServiceEditor` runtime/mapping field for example, we do some validation there).